src/DGtal/shapes/MeshHelpers.ih

```diff
@@ -0,0 +1,136 @@
+    trisurf.addVertex( *it );
+  for ( auto it = mesh.faceBegin(), itE = mesh.faceEnd(); it != itE; ++it )
+    {
+      typename Mesh<Point>::MeshFace face = *it;
+      for (unsigned int i = 1; i < face.size() - 1; i++ )
+        {
+          trisurf.addTriangle( face[ 0 ], face[ i ], face[ i+1 ] );
+        }
+    }
+  return trisurf.build();
+}
+
+template <typename Point>
+inline
+void
+DGtal::MeshHelpers::triangulatedSurface2Mesh
+( const TriangulatedSurface<Point>& trisurf,
+  Mesh<Point>& mesh )
+{
+  typedef typename TriangulatedSurface<Point>::Index Index;
+  for ( Index idx = 0; idx < trisurf.nbVertices(); ++idx )
+    mesh.addVertex( trisurf.position( idx ) );
+  for ( Index idx = 0; idx < trisurf.nbFaces(); ++idx )
+    {
+      auto vertices = trisurf.verticesAroundFace( idx );
+      mesh.addTriangularFace( vertices[ 0 ], vertices[ 1 ], vertices[ 2 ] );
+    }
+}
+
+template < typename DigitalSurfaceContainer,
+           typename CellEmbedder >
+inline
+void
+DGtal::MeshHelpers::digitalSurface2TriangulatedSurface
+    ( const DigitalSurface<DigitalSurfaceContainer>& dsurf,
+      const CellEmbedder& cembedder,
+      TriangulatedSurface<typename CellEmbedder::Value>& trisurf )
+{
+  BOOST_CONCEPT_ASSERT(( concepts::CCellEmbedder< CellEmbedder > ));
+  BOOST_CONCEPT_ASSERT(( concepts::CDigitalSurfaceContainer< DigitalSurfaceContainer > ));
+  typedef DigitalSurface< DigitalSurfaceContainer >    Surface;
+  typedef typename Surface::KSpace                     KSpace;
+  typedef typename Surface::Vertex                     Vertex;
+  typedef typename Surface::VertexRange                VertexRange;
+  typedef typename CellEmbedder::Value                 Point;
+  typedef typename TriangulatedSurface< Point >::Index Index;
+
+  trisurf.clear();
+  // Numbers all vertices and add them to the triangulated surface.
+  std::map<Vertex, Index> v_indices;
+  const KSpace & K = dsurf.container().space();
+  for ( auto it = dsurf.begin(), it_end = dsurf.end(); it != it_end; ++it )
+    {
+      const Vertex& v = *it;
+      v_indices[ v ] = trisurf.addVertex( cembedder( K.unsigns( v ) ) );
+    }
+
+  // Outputs closed faces.
+  auto faces = dsurf.allClosedFaces();
+  for ( auto itf = faces.begin(), itf_end = faces.end(); itf != itf_end; ++itf )
+    {
+      VertexRange vtcs = dsurf.verticesAroundFace( *itf );
+      if ( vtcs.size() == 3 )
+        trisurf.addTriangle( v_indices[ vtcs[ 0 ]   ],
+                             v_indices[ vtcs[ 1 ]   ],
+                             v_indices[ vtcs[ 2 ] ] );
+      else
+        { // We must add a vertex before triangulating.
+          Point barycenter;
+          for ( unsigned int i = 0; i < vtcs.size(); ++i )
+            barycenter += cembedder( K.unsigns( vtcs[ i ] ) );
+          barycenter /= vtcs.size();
+          Index idx = trisurf.addVertex( barycenter );
+          for ( unsigned int i = 0; i < vtcs.size(); ++i )
+            trisurf.addTriangle( v_indices[ vtcs[ i ]   ],
```

Comment:
Does choosing the first two vertices have an impact on the output triangulated surface ?
Are you sure that there is not overlapping triangles ?